### DISCUSSION TITLE
OSError: [E050] Can't find model 'en_core_web_sm'. It doesn't seem to be a Python package or a valid path to a data directory.

### DISCUSSION THREAD

**Discussion by tanyamole1** [explosion/spaCy#12638] [Help: Installation] [answered]
I'm trying to import spacy on Mac OS. I have tried following the guidance on https://spacy.io/usage for pip and condo  and https://www.projectpro.io/recipes/install-and-use-spacy-models. I cannot seem to understand what is happening. Could anyone please help explain what I'm doing wrong and how to fix this issue please? This is the error code in Jupyter notebook: 

Successfully installed en-core-web-sm-3.5.0
✔ Download and installation successful
You can now load the package via spacy.load('en_core_web_sm')
In [3]:

nlp = spacy.load('en_core_web_sm')
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
Cell In[3], line 1
----> 1 nlp = spacy.load('en_core_web_sm')

NameError: name 'spacy' is not defined

**Comment by danieldk** [comment]
Did you run

```
import spacy
```

in the notebook before

```
nlp = spacy.load('en_core_web_sm')
```

If so, could you give all the cells that you run in the notebook?

**Reply by tanyamole1** [reply]
Yes I did run those the, errors I get are: 

OSError                                   Traceback (most recent call last)
Cell In[2], line 1
----> 1 nlp = spacy.load('en_core_web_sm')

File ~/anaconda3/lib/python3.10/site-packages/spacy/__init__.py:54, in load(name, vocab, disable, enable, exclude, config)
     30 def load(
     31     name: Union[str, Path],
     32     *,
   (...)
     37     config: Union[Dict[str, Any], Config] = util.SimpleFrozenDict(),
     38 ) -> Language:
     39     """Load a spaCy model from an installed package or a local path.
     40 
     41     name (str): Package name or model path.
   (...)
     52     RETURNS (Language): The loaded nlp object.
     53     """
---> 54     return util.load_model(
     55         name,
     56         vocab=vocab,
     57         disable=disable,
     58         enable=enable,
     59         exclude=exclude,
     60         config=config,
     61     )

File ~/anaconda3/lib/python3.10/site-packages/spacy/util.py:449, in load_model(name, vocab, disable, enable, exclude, config)
    447 if name in OLD_MODEL_SHORTCUTS:
    448     raise IOError(Errors.E941.format(name=name, full=OLD_MODEL_SHORTCUTS[name]))  # type: ignore[index]
--> 449 raise IOError(Errors.E050.format(name=name))

OSError: [E050] Can't find model 'en_core_web_sm'. It doesn't seem to be a Python package or a valid path to a data directory.

It seems to be working now, I tried so many different ways to fix this I don't know which worked! Thank you for your offer of help

**Reply by danieldk** [reply]
This means that the `en_core_web_sm` model is not downloaded in the environment that you are using spaCy in.

**Reply by tanyamole1** [reply]
Can you provide me with the steps to resolve this please? VS Code is giving this error: 
Traceback (most recent call last):
  File "/Users/tanyamole/example.py", line 23, in <module>
    import spaCy #This statement should work if you have spaCy installed 
ModuleNotFoundError: No module named 'spaCy'

My terminal shows this:
============================== Info about spaCy ==============================

spaCy version    3.5.3                         
Location         /Users/tanyamole/anaconda3/lib/python3.10/site-packages/spacy
Platform         macOS-13.1-arm64-arm-64bit    
Python version   3.10.9                        
Pipelines        en_core_web_sm (3.5.0)

**Reply by danieldk** [reply, marked as answer]
Note that the module name is `spacy`, not `spaCy`, so you need to

```
import spacy
```

**Comment by tanyamole1** [comment]
Perhaps as I have now installed home-brew?

**Comment by tanyamole1** [comment]
I am wrong spacy isn't working in Jupyter notebook or VS Code. error for VS code is:
(base) tanyamole@Tanyas-iMac spacy_venv % /usr/bin/python3 /Users/tanyamole/Desktop/example.py
Traceback (most recent call last):
  File "/Users/tanyamole/Desktop/example.py", line 23, in <module>
    import spacy #This statement should work if you have spaCy installed 
ModuleNotFoundError: No module named 'spacy'
